plugins/kernel_v2/config/cloud_init.rb

```diff
@@ -49,6 +65,13 @@ def finalize!
       def validate(machine)
         errors = _detected_errors
 
+        if @type && @type != DEFAULT_CONFIG_TYPE
```

Comment:
Will other types be supported at some point as well?

Reply:
It sounds like it's a maybe, but for the most part this type is to make it so that a new type _could_ exist in the future without breaking older Vagrantfiles if we do end up introducing a new kind of config. So for now there's really only one type, thus the check here! :smile: